### PR TITLE
Feature/hopper support concept

### PR DESCRIPTION
**Pull request by Th3Shadowbroker**
As mentioned in #14. Looking forward to your feedback.

- Added hopper support (in-/output)

### CODE REVIEW

**Comment by InventivetalentDev** [comment]
Really nice implementation, thanks! :)
Couple issues I've noticed testing it a bit:
* The hopper transfers are not triggered for newly placed bookshelves
* Non-book items can be transferred into bookshelves at the moment, should probably check that the same way [as in the inventory events](https://github.com/InventivetalentDev/Bookshelves/blob/master/src/org/inventivetalent/bookshelves/ShelfListener.java#L95) if only books are allowed via the config

**Comment by Th3Shadowbroker** [comment]
Thx for the feedback. I'll have a look at it 🙂 
I totally missed checking newly placed block *facepalm*. Probably because I used the ones I already had placed in my test-world 😅 I'll also readd the material validation for the items. Forgot about it after I discarded my first approach ^^

File: src/org/inventivetalent/bookshelves/Bookshelves.java [hunk 1]
```diff
@@ -64,6 +65,7 @@ public void onEnable() {
 		if (getConfig().contains("disabledWorlds")) { disabledWorlds.addAll(getConfig().getStringList("disabledWorlds")); }
 		onlyBooks = getConfig().getBoolean("onlyBooks", true);
 		checkRestrictions = getConfig().getBoolean("restrictions.enabled");
+		hopperSupport = getConfig().getBoolean("onlyBooks");
```
Comment (InventivetalentDev) [inline src/org/inventivetalent/bookshelves/Bookshelves.java]: Correct key? 🤔

Comment (Th3Shadowbroker) [reply src/org/inventivetalent/bookshelves/Bookshelves.java]: shhhh https://github.com/InventivetalentDev/Bookshelves/pull/17/commits/82e6e29efa2a78aa6bc8f4698a53982d87dd4b0d

**Pull request merged**